tests/e2e/csi_cns_telemetry.go

```diff
@@ -0,0 +1,207 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	fnodes "k8s.io/kubernetes/test/e2e/framework/node"
+	fpv "k8s.io/kubernetes/test/e2e/framework/pv"
+)
+
+var _ = ginkgo.Describe("[csi-block-vanilla] [csi-file-vanilla] CNS-CSI Cluster Distribution Telemetry", func() {
+	f := framework.NewDefaultFramework("csi-cns-telemetry")
+	var (
+		client       clientset.Interface
+		namespace    string
+		scParameters map[string]string
+		datastoreURL string
+	)
+
+	ginkgo.BeforeEach(func() {
+		bootstrap()
+		client = f.ClientSet
+		namespace = getNamespaceToRunTests(f)
+		scParameters = make(map[string]string)
+		datastoreURL = GetAndExpectStringEnvVar(envSharedDatastoreURL)
+		nodeList, err := fnodes.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		//Reset the cluster distribution value to original value
+		isSet := setClusterDistributionValue4Vanilla(ctx, client, vanillaClusterDistribution)
+		gomega.Expect(isSet).To(gomega.BeTrue())
+	})
+
+	ginkgo.AfterEach(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		//Reset the cluster distribution value to original value
+		isSet := setClusterDistributionValue4Vanilla(ctx, client, vanillaClusterDistribution)
+		gomega.Expect(isSet).To(gomega.BeTrue())
+	})
+
+	/*
+		Test to verify cluster distribution set in PVC is being honored during volume creation.
+		Steps
+			1. Create StorageClass
+			2. Create PVC with valid disk size
+			3. Expect PVC to pass and cluster distribution value set
+			4. Update cluster-distribution value
+			5. Create another PVC with valid disk size
+			6. Expect PVC to pass and cluster distribution value set to latest update
+			7. Expect the old PVC to reflect the latest cluster-distribution value
+	*/
+
+	// Test for cluster-distribution value presence
+	ginkgo.It("Verify dynamic provisioning of pvc has cluster-distribution value updated", func() {
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		ginkgo.By("Invoking cluster-distribution for volume basic test")
```

Comment:
ginkgo.By should describe the test youre about to execute. I think this should be "Creating a PVC"